src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -204,16 +205,30 @@ public static Result canEntitySpawn(Mob entity, LevelAccessor world, double x, d
         return event.getResult();
     }
 
+    // TODO 1.18.2/1.19 remove in favor of LevelAccessor & CustomSpawnRules version
+    @Deprecated
     public static boolean canEntitySpawnSpawner(Mob entity, Level world, float x, float y, float z, BaseSpawner spawner)
+    {
+        return canEntitySpawnSpawner(entity, world, x, y, z, spawner, Optional.empty());
+    }
+
+    public static boolean canEntitySpawnSpawner(Mob entity, LevelAccessor world, float x, float y, float z, BaseSpawner spawner, Optional<SpawnData.CustomSpawnRules> spawnRules)
     {
         Result result = canEntitySpawn(entity, world, x, y, z, spawner, MobSpawnType.SPAWNER);
         if (result == Result.DEFAULT)
-            return entity.checkSpawnRules(world, MobSpawnType.SPAWNER) && entity.checkSpawnObstruction(world); // vanilla logic (inverted)
+            return !(spawnRules.isEmpty() && !entity.checkSpawnRules(world, MobSpawnType.SPAWNER) || !entity.checkSpawnObstruction(world)); // vanilla logic (inverted)
```

Comment:
This is the same as `!spawnRules.isEmpty() || entity.checkSpawnRules(world, MobSpawnType.SPAWNER) && entity.checkSpawnObstruction(world);` just a lot less readable

Reply:
In this case, since the main goal of this line is to mimic vanilla code, I feel that matching vanilla code is more important than readability for the sake of future maintenance. 

May be worth noting that the `||` part would need to be wrapped in parentheses to keep the order of operations intact.

Reply:
> I feel that matching vanilla code is more important than readability for the sake of future maintenance. 

We don't. 

Why would you ever prioritise keeping the dodgy decompiler output over something much simpler and easier for us to maintain in the future?

That seems like an intentional FU to the people who have to fix this stuff when it breaks.